src/Cli/dotnet/ShellShim/ShellShimRepository.cs

```diff
@@ -134,38 +162,45 @@ private class RootObject
         public StartupOptions startupOptions { get; set; }
     }
 
-    private bool ShimExists(ToolCommandName commandName)
+    private bool ShimExists(ToolCommand toolCommand)
     {
-        return GetShimFiles(commandName).Any(p => _fileSystem.File.Exists(p.Value));
+        return GetShimFiles(toolCommand).Any(p => _fileSystem.File.Exists(p.Value));
     }
 
-    private IEnumerable<FilePath> GetShimFiles(ToolCommandName commandName)
+    private IEnumerable<FilePath> GetShimFiles(ToolCommand toolCommand)
     {
-        yield return GetShimPath(commandName);
+        yield return GetShimPath(toolCommand);
     }
 
-    private FilePath GetShimPath(ToolCommandName commandName)
+    private FilePath GetShimPath(ToolCommand toolCommand)
     {
         if (OperatingSystem.IsWindows())
         {
-            return _shimsDirectory.WithFile(commandName.Value + ".exe");
+            if (toolCommand.Runner == "dotnet")
```

Comment:
Do we need to support Powershell at all?

Reply:
Good point, but testing just now in powershell it seems like it will resolve .cmd files that are in the path:

```
PS C:\Users\Daniel> ToolTest
Hello, Tool!
PS C:\Users\Daniel> (Get-Command ToolTest).Path
C:\Users\Daniel\.dotnet\tools\ToolTest.cmd
```

So it looks like we don't need to do anything more.